x/blob/keeper/keeper.go

```diff
@@ -61,9 +61,12 @@ func (k Keeper) PayForBlob(goCtx context.Context, msg *types.MsgPayForBlob) (*ty
 	gasToConsume := uint64(shares.MsgSharesUsed(int(msg.BlobSize)) * GasPerMsgShare)
 	ctx.GasMeter().ConsumeGas(gasToConsume, payForBlobGasDescriptor)
 
-	ctx.EventManager().EmitEvent(
+	err := ctx.EventManager().EmitTypedEvent(
 		types.NewPayForBlobEvent(sdk.AccAddress(msg.Signer).String(), msg.GetBlobSize()),
 	)
+	if err != nil {
+		return &types.MsgPayForBlobResponse{}, nil
```

Comment:
```suggestion
		return &types.MsgPayForBlobResponse{}, err
```

Reply:
Whoops, thanks for catching this. I'll change it right away